docs/content/docs/internals/job_scheduling.md

```diff
@@ -95,3 +95,22 @@ For that reason, the execution of an ExecutionVertex is tracked in an {{< gh_lin
 {{< img src="/fig/state_machine.svg" alt="States and Transitions of Task Executions" width="50%" >}}
 
 {{< top >}}
+
+## Repeatable Resource Cleanup Strategy
+
+Once a job has reached a terminally global state of either finished, failed or cancelled, the
+resources associated with the job are then cleaned up. This is done via a repeatable
+resource cleanup strategy, in which failures to clean up a resource result retries separated by
+an exponentially increasing delay, within configured bounds.
+
+{{< img src="/fig/repeatable_cleanup.png" alt="Repeatable resource cleanup across a failover event" >}}
+
+Determining which jobs are globally terminated but still need to be cleaned up across
+a failover event is done by determining whether an entry for the job exists in the JobResultStore
+and whether that entry is dirty (and thus needs resource cleanup) or clean (and thus does not
+need any further cleanup).
```

Comment:
Hm, initially I thought it's a good idea and we should cross-reference the JobResultStore docs here. But then again, I was wondering whether we should, because the `JobResultStore` is mostly relevant in HA mode. But the retries would also work in non-HA mode and might confuse users in that case.

Hence, I was wondering whether we should handle this paragraph as not relevant for the user.